app/models/grant_wallet.rb

```diff
@@ -0,0 +1,23 @@
+class GrantWallet < ApplicationRecord
+    belongs_to :user
+    belongs_to :event
+
+    def grant_to(camp, requested_grants)
+        max_grants_per_user_per_dream = Rails.configuration.x.firestarter_settings['max_grants_per_user_per_dream']
+        num_grants = [requested_grants, self.grants_left, max_grants_per_user_per_dream].min
+
+        til_funded = camp.maxbudget - num_grants
+        num_grants = til_funded if num_grants > til_funded
+        
+        camp.grants.build(user: self.user, amount: num_grants)
+        camp.assign_attributes(
+            minfunded:   (camp.grants_received + num_grants) >= camp.minbudget,
```

Comment:
A little note here: We'd be way better off taking these columns off of the camp model entirely, and simply writing methods to mimic what they do; there's no need to store booleans like this in the DB if they can be inferred simply from other info on the model.
```
# camp.rb
def minfunded
  grants_received >= minbudget
end

def fullyfunded
  grants_received >= maxbudget
end
```

Also note that a slightly more idiomatic naming might be `fully_funded?` and `min_funded?`